catroid/src/main/java/org/catrobat/catroid/common/BroadcastSequenceMap.java

```diff
@@ -28,43 +28,43 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public final class BroadcastSequenceMap {
-	private static Map<String, HashMap<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
+public class BroadcastSequenceMap {
+	protected Map<String, Map<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
 
-	private BroadcastSequenceMap() {
-		throw new AssertionError();
+	public boolean containsKey(String key, String sceneName) {
+		return broadcastSequenceMap.containsKey(sceneName) && broadcastSequenceMap.get(sceneName).containsKey(key);
 	}
 
-	public static boolean containsKey(String key, String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
-			return false;
+	public ArrayList<SequenceAction> get(String key, String sceneName) {
+		Map<String, ArrayList<SequenceAction>> map = broadcastSequenceMap.get(sceneName);
+		if (map == null) {
+			map = new HashMap<>();
+			broadcastSequenceMap.put(sceneName, map);
+		}
+		if (!map.containsKey(key)) {
+			map.put(key, new ArrayList<SequenceAction>());
 		}
-		return BroadcastSequenceMap.broadcastSequenceMap.get(sceneName).containsKey(key);
+		return map.get(key);
 	}
 
-	public static ArrayList<SequenceAction> get(String key, String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
-			return null;
-		}
-		return BroadcastSequenceMap.broadcastSequenceMap.get(sceneName).get(key);
+	public boolean add(String sceneName, String key, SequenceAction value) {
+		return get(key, sceneName).add(value);
 	}
 
-	public static ArrayList<SequenceAction> put(String sceneName, String key, ArrayList<SequenceAction> value) {
-		if (!broadcastSequenceMap.containsKey(sceneName)) {
-			HashMap<String, ArrayList<SequenceAction>> map = new HashMap<>();
-			broadcastSequenceMap.put(sceneName, map);
+	public void remove(String key, String sceneName) {
+		if (containsKey(key, sceneName)) {
+			broadcastSequenceMap.get(sceneName).remove(key);
 		}
-		return BroadcastSequenceMap.broadcastSequenceMap.get(sceneName).put(key, value);
 	}
 
-	public static void clear() {
-		BroadcastSequenceMap.broadcastSequenceMap.clear();
+	public void clear() {
+		broadcastSequenceMap.clear();
 	}
 
-	public static void clear(String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
+	public void clear(String sceneName) {
+		if (!broadcastSequenceMap.containsKey(sceneName)) {
```

Comment:
please change to:
if (broadcastSequenceMap.containsKey(sceneName)) {
    broadcastSequenceMap.get(sceneName).clear();
}
its a lot easier to read...